extensions/openshift-client/deployment/src/main/java/io/quarkus/openshift/client/deployment/OpenShiftClientProcessor.java

```diff
@@ -0,0 +1,137 @@
+                .stream()
+                .peek(c -> {
+                    // we need to make sure that the Jackson extension does not try to fully register the model classes
+                    // since we are going to register them weakly
+                    ignoredJsonDeserializationClasses.produce(new IgnoreJsonDeserializeClassBuildItem(c.name()));
+                })
+                .map(c -> c.name().toString())
+                .filter(c -> !watchedClasses.contains(c))
+                .toArray(String[]::new);
+        reflectiveClasses.produce(ReflectiveClassBuildItem
+                .builder(modelClasses).weak(true).methods(true).fields(false).build());
+
+        // we also ignore some classes that are annotated with @JsonDeserialize that would force the registration of the entire model
+        ignoredJsonDeserializationClasses.produce(
+                new IgnoreJsonDeserializeClassBuildItem(DotName.createSimple("io.fabric8.kubernetes.api.model.KubeSchema")));
+        ignoredJsonDeserializationClasses.produce(
+                new IgnoreJsonDeserializeClassBuildItem(
+                        DotName.createSimple("io.fabric8.kubernetes.api.model.KubernetesResourceList")));
+        ignoredJsonDeserializationClasses.produce(new IgnoreJsonDeserializeClassBuildItem(KUBERNETES_RESOURCE));
+
+        final String[] doneables = combinedIndexBuildItem.getIndex()
+                .getAllKnownImplementors(DotName.createSimple("io.fabric8.kubernetes.api.model.Doneable"))
+                .stream()
+                .map(c -> c.name().toString())
+                .toArray(String[]::new);
+        reflectiveClasses.produce(ReflectiveClassBuildItem.weakClass(doneables));
+
+        final String[] deserializerClasses = combinedIndexBuildItem.getIndex()
+                .getAllKnownSubclasses(DotName.createSimple("com.fasterxml.jackson.databind.JsonDeserializer"))
+                .stream()
+                .map(c -> c.name().toString())
+                .filter(s -> s.startsWith("io.fabric8.kubernetes"))
+                .toArray(String[]::new);
+        reflectiveClasses.produce(new ReflectiveClassBuildItem(true, false, deserializerClasses));
+
+        final String[] serializerClasses = combinedIndexBuildItem.getIndex()
+                .getAllKnownSubclasses(DotName.createSimple("com.fasterxml.jackson.databind.JsonSerializer"))
+                .stream()
+                .map(c -> c.name().toString())
+                .filter(s -> s.startsWith("io.fabric8.kubernetes"))
+                .toArray(String[]::new);
+        reflectiveClasses.produce(new ReflectiveClassBuildItem(true, false, serializerClasses));
+
+        reflectiveClasses
+                .produce(new ReflectiveClassBuildItem(true, false, "io.fabric8.kubernetes.api.model.IntOrString"));
+        reflectiveClasses
+                .produce(new ReflectiveClassBuildItem(true, false, "io.fabric8.kubernetes.internal.KubernetesDeserializer"));
```

Comment:
Shouldn't part of this be shared to avoid having to fix bugs in both extension? Most of this doesn't look specific to OpenShift?

Reply:
If there is a way it would be nice to use it, as more client extensions could be added in the future. Personally I'd love to see `kantive` and `tekton`. So that would be awesome!

Reply:
+1 for what both Guillaume and Ioannis said above

Reply:
> Shouldn't part of this be shared to avoid having to fix bugs in both extension? Most of this doesn't look specific to OpenShift?

this code is not duplicated anymore.
are you suggesting to create a base kubernetes extension that includes part of the `KubernetesClientProcessor`? then, it would have to depend on the fabric8 kubernetes-client library, right; because the referred classes such as `io.fabric8.kubernetes.api.model.IntOrString` come from that library. 
there are not `knative` or `tekton` libraries today? if there were, you idea is to get them to share those common classes? then we are talking about some code reuse in the fabric8 project with support for `knative` and `tekton` modules, before even talking about some code reuse in `quarkus`?
what exactly do you think we can do at this point here in `quarkus`?